### PR TITLE
feat(replay): Iterate on the details page header

### PR DESCRIPTION
**Pull request by ryan953**
Iterate on the details page header to make it closer to the mocks

**Before:**
Screen Shot 2022-04-28 at 10 21 08 PM

**After:**
Screen Shot 2022-04-29 at 9 42 21 AM

Also updated the <UserBadge> on the list page to match what's going on in the details page.

### CODE REVIEW

File: static/app/views/replays/replays.tsx [hunk 1]
```diff
@@ -106,11 +117,11 @@ function Replays(props: Props) {
...
-            username: replay['user.display'],
-            id: replay['user.display'],
-            ip_address: replay['user.display'],
-            name: replay['user.display'],
-            email: replay['user.display'],
+            username: replay['user.username'] ?? '',
+            id: replay['user.id'] ?? '',
+            ip_address: replay['user.ip_address'] ?? '',
+            name: replay['user.name'] ?? '',
+            email: replay['user.email'] ?? '',
```
Comment (ryan953) [inline static/app/views/replays/replays.tsx:124]: before the generated avatar image was something like:
Screen Shot 2022-04-28 at 3 26 47 PM

The color and digits are based on the first digit in the ip address :(

Now we pass in any user data that we have fetched, including the ip_address. Now if only the `ip_address` field is present (this is the case for all our test data) the avatar image will instead always be:
Screen Shot 2022-04-28 at 3 28 49 PM

This is to match with the fields we have in the details page, so the avatar images are the same.

File: static/app/views/replays/replays.tsx [hunk 2]
```diff
@@ -247,12 +258,6 @@ const StyledPanelItem = styled('div')`
   margin-top: ${space(0.75)};
 `;
 
-const ReplayUserBadge = styled(UserBadge)`
-  font-size: ${p => p.theme.fontSizeLarge};
-  font-weight: 400;
-  line-height: 1.2;
-`;
-
```
Comment (ryan953) [inline static/app/views/replays/replays.tsx:255]: Turned out this wasn't doing anything anymore

All the props are overridden by the two spans that render (screen shot is for `<StyledEmail>`, but `<StyledName>` does the same):
Screen Shot 2022-04-28 at 3 20 08 PM

File: static/app/views/replays/detail/detailLayout.tsx [hunk 3]
```diff
@@ -2,41 +2,69 @@ import React from 'react';
...
-                {label: t('Replay Details')}, // TODO(replay): put replay ID or something here
               ]}
             />
+            <ButtonWrapper>
+              <Button
+                title={t('Send us feedback via email')}
+                href="mailto:replay-feedback@sentry.io?subject=Replay Details Feedback"
+              >
+                {t('Give Feedback')}
+              </Button>
```
Comment (billyvg) [inline static/app/views/replays/detail/detailLayout.tsx:63]: We could use this when the PR gets merged https://github.com/getsentry/sentry/pull/33858

**Review by billyvg — APPROVED** [review]

File: static/app/views/replays/detail/detailLayout.tsx [hunk 4]
```diff
@@ -2,41 +2,69 @@ import React from 'react';
...
+                    </React.Fragment>
+                  ),
+                },
+                {
+                  label: event
+                    ? generateEventSlug({
+                        project: event.projectSlug,
+                        id: event.id,
+                      })
+                    : t('Replay'),
```
[the remark this answers is not in the document]
Comment (ryan953) [reply static/app/views/replays/detail/detailLayout.tsx:53]: I can agree with that easily. I'll drop it.

We had the comment `// TODO(replay): put replay ID or something here` so I thought it was a good time to try out what that looks like. 

The downsides of having the `id` in the breadcrumb like this is that it's trimmed!

File: static/app/views/replays/detail/detailLayout.tsx (hunk 3 above)
Comment (ryan953) [reply static/app/views/replays/detail/detailLayout.tsx:63]: Ooooh, that's the component to use for sure.

**Pull request merged**